src/apig_wsgi.py

```diff
@@ -55,7 +55,8 @@ def get_environ(event, binary_support):
     else:
         body = body.encode("utf-8")
     params = event.get("queryStringParameters") or {}
-
+    # decoding first to prevent double encoding
+    params = {k: unquote(v) for k, v in params.items()}
```

Comment:
You're only unquoting values -  did you verify the issue doesn't exist in keys?

Reply:
My thinking was that you would hopefully never have an application that has keys like that :). Probably not a safe assumption though.

Reply:
A URL like `example.com/?+123` should have a key with an empty string as the value